Update documentation of reset behavior of state variables in modes (#67)
lf-lang/website-lingua-franca: packages/documentation/copy/en/topics/Modal Models.md

In case a mode is entered with the reset behavior:

* all contained modal reactors are reset to their initial mode (recursively),
* all contained timers are reset and start again awaiting their initial offset,
* all contained state variables that are marked for automatic reset are reset to their initial value,

> * the `reset` triggers and runs all associated contained reactions, and

Reviewer: ```suggestion
* any contained reactions triggered by `reset` are executed, and
```
